### PR TITLE
Scalebar polls its data

### PR DESCRIPTION
**Pull request by BarAgam**
Scalebar now polls the MapView and update its text at 60 FPS, just like the Compass.
In fact, Compass and Scalebar now share the 60FPS polling logic.

### CODE REVIEW

File: app/src/main/java/com/teamagam/gimelgimel/app/map/esri/plugins/Compass.java [hunk 1]
```diff
@@ -46,12 +41,11 @@
     private final MapView mMapView;
 
     private float mAngle;
-    private Subscription mRefreshSubscription;
-    private boolean mIsRunning;
     private int mHeight;
     private int mWidth;
     private int mCenterX;
     private int mCenterY;
+    private UIUpdatePoller mCompassUIUpdatePoller;
```
Comment (teamagam5) [inline app/src/main/java/com/teamagam/gimelgimel/app/map/esri/plugins/Compass.java]: rename to mUIUpdatePoller

File: app/src/main/java/com/teamagam/gimelgimel/app/map/esri/plugins/Compass.java [hunk 2]
```diff
@@ -131,8 +118,20 @@ private void rotateMatrix() {
...
+        CompassUIUpdatePoller() {
+            super(Schedulers.computation());
+        }
+
+        @Override
+        protected void periodicalAction() {
+            setRotationAngle(mMapView.getRotationAngle());
+        }
+
+        private void setRotationAngle(double angle) {
```
Comment (teamagam5) [inline app/src/main/java/com/teamagam/gimelgimel/app/map/esri/plugins/Compass.java]: why was this method moved to the inner class?

File: app/src/main/java/com/teamagam/gimelgimel/app/map/esri/plugins/ScaleBar.java [hunk 3]
```diff
@@ -95,4 +66,22 @@ private String format(double scale) {
...
+
+    private class ScaleBarUIUpdatePoller extends UIUpdatePoller {
+
+        ScaleBarUIUpdatePoller() {
+            super(Schedulers.computation());
+        }
+
+        @Override
+        protected void periodicalAction() {
+            if (mMapView != null) {
```
Comment (teamagam5) [inline app/src/main/java/com/teamagam/gimelgimel/app/map/esri/plugins/ScaleBar.java]: Better check it on start, so when view is not set you don't even start the poller.

File: app/src/main/java/com/teamagam/gimelgimel/app/map/esri/plugins/UIUpdatePoller.java [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+    public UIUpdatePoller(Scheduler workScheduler) {
+        mWorkScheduler = workScheduler;
+        mSubscription = null;
+        mIsRunning = false;
+    }
+
+    public void start() {
+        if (!mIsRunning) {
+            mSubscription =
+                    Observable.interval(Constants.COMPASS_REFRESH_RATE_MS, TimeUnit.MILLISECONDS)
```
Comment (teamagam5) [inline app/src/main/java/com/teamagam/gimelgimel/app/map/esri/plugins/UIUpdatePoller.java]: rename COMPASS_REFRESH_RATE_MS

File: app/src/main/java/com/teamagam/gimelgimel/app/map/esri/plugins/UIUpdatePoller.java [hunk 5]
```diff
@@ -0,0 +1,44 @@
...
+import rx.Subscription;
+
+abstract class UIUpdatePoller {
+
+    private final Scheduler mWorkScheduler;
+    private Subscription mSubscription;
+    private boolean mIsRunning;
+
+    public UIUpdatePoller(Scheduler workScheduler) {
+        mWorkScheduler = workScheduler;
```
Comment (teamagam5) [inline app/src/main/java/com/teamagam/gimelgimel/app/map/esri/plugins/UIUpdatePoller.java:20]: Why let the extending classes pass this scheduler? use Schedulers.computation() in here and save them this decision... They both use it.

File: app/src/main/java/com/teamagam/gimelgimel/app/map/esri/plugins/ScaleBar.java [hunk 6]
```diff
@@ -95,4 +66,22 @@ private String format(double scale) {
...
+                updateScale(mMapView.getScale());
+            }
+        }
+
+        private void updateScale(double scale) {
```
Comment (teamagam5) [inline app/src/main/java/com/teamagam/gimelgimel/app/map/esri/plugins/ScaleBar.java]: once again, why was this method moved to the inner class?

File: app/src/main/java/com/teamagam/gimelgimel/app/map/esri/EsriGGMapView.java [hunk 7]
```diff
@@ -95,17 +93,15 @@ public EsriGGMapView(Context context, AttributeSet attrs) {
     @Override
     public void unpause() {
         super.unpause();
-        if (mCompass != null) {
-            mCompass.start();
-        }
+        startPlugin(mCompass);
+        startPlugin(mScaleBar);
```
Comment (teamagam5) [inline app/src/main/java/com/teamagam/gimelgimel/app/map/esri/EsriGGMapView.java:97]: I'm not sure it's needed but these can be grouped under "startPlugins(). And the same for the stop...

**Review by teamagam5 — APPROVED** [review]

File: app/src/main/java/com/teamagam/gimelgimel/app/map/esri/plugins/Compass.java (hunk 1 above)
Comment (BarAgam) [reply app/src/main/java/com/teamagam/gimelgimel/app/map/esri/plugins/Compass.java]: K

File: app/src/main/java/com/teamagam/gimelgimel/app/map/esri/plugins/ScaleBar.java (hunk 3 above)
Comment (BarAgam) [reply app/src/main/java/com/teamagam/gimelgimel/app/map/esri/plugins/ScaleBar.java]: K

File: app/src/main/java/com/teamagam/gimelgimel/app/map/esri/plugins/UIUpdatePoller.java (hunk 5 above)
Comment (BarAgam) [reply app/src/main/java/com/teamagam/gimelgimel/app/map/esri/plugins/UIUpdatePoller.java:20]: Theoretically, it helps me create unit-tests for that class

File: app/src/main/java/com/teamagam/gimelgimel/app/map/esri/plugins/UIUpdatePoller.java (hunk 4 above)
Comment (BarAgam) [reply app/src/main/java/com/teamagam/gimelgimel/app/map/esri/plugins/UIUpdatePoller.java]: K

**Pull request merged**